defi/src/api2/cron-task/dimensions.ts

```diff
@@ -125,23 +515,66 @@ async function run() {
       protocols: {},
     }
     const adapterData = allCache[adapterType]
+    // init per-protocol backfill flags map
+    if (!adapterData.blHistoryFetched) adapterData.blHistoryFetched = {}
 
     await pullChangedFromDBAndAddToCache()
 
-
     async function pullChangedFromDBAndAddToCache() {
       let lastUpdated = allCache[adapterType].lastUpdated ? allCache[adapterType].lastUpdated - 1 * 60 * 60 : 0 // 1 hour ago
+      
       const results = await getAllItemsUpdatedAfter({ adapterType, timestamp: lastUpdated })
 
+      const protocolsWithBlDataInRecentUpdates = new Set<string>()
+      
       results.forEach((result: any) => {
-        const { id, timestamp, data, timeS, } = result
-        roundVaules(data)
-        if (!adapterData.protocols[id]) adapterData.protocols[id] = {
-          records: {}
+        const { id, timestamp, data, timeS } = result
+
+        const bl = (result as any).bl  ?? (result as any).breakdownLabel ?? (result as any).data?.bl  ?? (result as any).data?.breakdownLabel
+
+        if (bl) {
+          protocolsWithBlDataInRecentUpdates.add(id)
         }
-        adapterData.protocols[id].records[timeS] = { ...data, timestamp, }
+
+        roundVaules(data)
+
+        if (!adapterData.protocols[id]) adapterData.protocols[id] = { records: {} }
+
+        const finalRecord: any = { ...data, timestamp }
+
+        if (bl)  { finalRecord.bl  = bl;  finalRecord.breakdownLabel = bl }
+
+        adapterData.protocols[id].records[timeS] = finalRecord
       })
 
+      // backfill full history ONLY for protocols that newly emitted bl and have not been backfilled
+      const toBackfill = Array.from(protocolsWithBlDataInRecentUpdates)
```

Comment:
no, it is much simpler to reset the cache once than do this, please remove this block of code